cas-server-webapp/src/main/webapp/WEB-INF/view/jsp/protocol/3.0/casServiceValidationSuccess.jsp

```diff
@@ -49,9 +49,6 @@
                         <cas:${fn:escapeXml(attr.key)}>${fn:escapeXml(attrval)}</cas:${fn:escapeXml(attr.key)}>
                     </c:forEach>
                 </c:forEach>
-                <cas:isFromNewLogin>${fn:escapeXml(isFromNewLogin)}</cas:isFromNewLogin>
```

Comment:
From the JSP perspective, everything is considered an attribute because they are all part of the "attributes" block.

From the CAS perspective, authN attributes are of course separated from principal attributes. The code will get all user attributes, and will add authN attributes to the same collection, and the result is passed onto the JSP view. What this means then is that if no user attributes are released, the final collection size is greater than zero because we have always authN attributes to pass. (Previously user-attributes=0 meant no attributes at all)